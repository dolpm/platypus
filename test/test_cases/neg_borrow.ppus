/* try to mutably borrow a borrowed variable */
pipe main |> [] |> [] |> unit {
  int x <| 0;
  {
    &int y <| &x;
    &int z <| &x;
    ~&int a <| ~&x;
  }
  |> ();
}