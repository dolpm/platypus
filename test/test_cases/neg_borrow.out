Fatal error: exception Failure("variable x is already immutably borrowed, thus it cant be mutably borrowed.")
